services/rds/rds.go

```diff
@@ -399,12 +399,14 @@ func (svc *Service) addMySQLdExporter(ctx context.Context, tx *reform.TX, servic
 		return errors.WithStack(err)
 	}
 
-	// check connection
+	// check connection and a number of tables
+	var tableCount int
 	dsn := agent.DSN(service)
 	db, err := sql.Open("mysql", dsn)
 	if err == nil {
-		err = db.PingContext(ctx)
+		err = db.QueryRowContext(ctx, "SELECT COUNT(*) FROM information_schema.tables").Scan(&tableCount)
 		db.Close()
+		agent.MySQLDisableTablestats = pointer.ToBool(tableCount > 1000)
```

Comment:
I don't know why we have so high limit, tablestat creates huge traffic and significant delays even on 200 tables.
anyway, feel free to merge!

Reply:
That's the same limit as in pmm-admin.

Currently, we do not provide UI for setting this number, but we store disabling flag in the database when we add an instance, so our support can explain to a customer how to switch it.